omnipaxos_core/src/sequence_paxos/leader.rs

```diff
@@ -393,8 +459,11 @@ where
             }
         }
         self.internal_storage
-            .set_accepted_round(self.leader_state.n_leader);
-        self.internal_storage.set_decided_idx(decided_idx);
+            .set_accepted_round(self.leader_state.n_leader)
+            .expect("storage error");
+        self.internal_storage
+            .set_decided_idx(decided_idx)
+            .expect("storage error");
```

Comment:
Same here, I think we can first set these and possibly rollback if snapshot or entries fail later.